pkg/servicemanager/manager.go

```diff
@@ -121,7 +125,7 @@ func (m *ServiceManager) asyncRun(ctx context.Context, service Service) (<-chan
 			svcStart := time.Now()
 			go func() {
 				<-ready
-				klog.InfoS("SERVICE READY", "service", service.Name(), "since-start", time.Since(svcStart))
+				m.startRec.ServiceReady(service.Name(), service.Dependencies(), svcStart)
```

Comment:
I believe this place is susceptible to a race condition.
When last service is `ready`, microshift becomes ready (in different thread) and emits the log message. And this happens before `.ServiceReady()` is called.
So the start up log we're emitting misses one service:

```
Dec 11 10:18:48 el94-base-el94-src-upgrade-ok-host1 microshift[1281]: ??? I1211 10:18:48.365552    1281 startuprecorder.go:69] "MICROSHIFT READY" since-start="23.915729142s"
Dec 11 10:18:48 el94-base-el94-src-upgrade-ok-host1 microshift[1281]: ??? I1211 10:18:48.365836    1281 startuprecorder.go:85] Startup data: {"services":[{"name":"network-configuration","dependencies":[],"start":"2024-12-11T10:18:25.234412309-05:00","ready":"2024-12-11T10:18:25.235467327-05:00","timeToReady":1055004},{"name":"sysconfwatch-controller","dependencies":[],"start":"2024-12-11T10:18:25.235064878-05:00","ready":"2024-12-11T10:18:25.235689348-05:00","timeToReady":624458},{"name":"etcd","dependencies":[],"start":"2024-12-11T10:18:25.235495816-05:00","ready":"2024-12-11T10:18:28.249009191-05:00","timeToReady":3013513372},{"name":"kube-apiserver","dependencies":["etcd","network-configuration"],"start":"2024-12-11T10:18:28.249062951-05:00","ready":"2024-12-11T10:18:31.254130427-05:00","timeToReady":3005067464},{"name":"openshift-crd-manager","dependencies":["kube-apiserver"],"start":"2024-12-11T10:18:31.254167974-05:00","ready":"2024-12-11T10:18:31.960607578-05:00","timeToReady":706439593},{"name":"openshift-default-scc-manager","dependencies":["kube-apiserver","openshift-crd-manager"],"start":"2024-12-11T10:18:31.961504478-05:00","ready":"2024-12-11T10:18:32.071371871-05:00","timeToReady":109867393},{"name":"microshift-mdns-controller","dependencies":["openshift-default-scc-manager"],"start":"2024-12-11T10:18:32.071405973-05:00","ready":"2024-12-11T10:18:32.072603259-05:00","timeToReady":1197275},{"name":"kube-scheduler","dependencies":["kube-apiserver"],"start":"2024-12-11T10:18:31.254150522-05:00","ready":"2024-12-11T10:18:36.273814284-05:00","timeToReady":5019663750},{"name":"kube-controller-manager","dependencies":["kube-apiserver"],"start":"2024-12-11T10:18:31.25554598-05:00","ready":"2024-12-11T10:18:36.275234611-05:00","timeToReady":5019688619},{"name":"route-controller-manager","dependencies":["kube-apiserver","openshift-crd-manager"],"start":"2024-12-11T10:18:31.960646879-05:00","ready":"2024-12-11T10:18:37.117336594-05:00","timeToReady":5156689716},{"name":"infrastructure-services-manager","dependencies":["kube-apiserver","openshift-crd-manager","route-controller-manager"],"start":"2024-12-11T10:18:37.11751121-05:00","ready":"2024-12-11T10:18:38.349272468-05:00","timeToReady":1231761257},{"name":"cluster-policy-controller","dependencies":["kube-apiserver","infrastructure-services-manager"],"start":"2024-12-11T10:18:38.349320859-05:00","ready":"2024-12-11T10:18:38.351420427-05:00","timeToReady":2099557},{"name":"kustomizer","dependencies":["kube-apiserver"],"start":"2024-12-11T10:18:38.358311994-05:00","ready":"2024-12-11T10:18:38.361308942-05:00","timeToReady":2996936},{"name":"version-manager","dependencies":["kube-apiserver"],"start":"2024-12-11T10:18:38.357527921-05:00","ready":"2024-12-11T10:18:38.368760181-05:00","timeToReady":11232248},{"name":"microshift-loadbalancer-service-controller","dependencies":["network-configuration","kube-apiserver","infrastructure-services-manager"],"start":"2024-12-11T10:18:38.350706899-05:00","ready":"2024-12-11T10:18:38.452860531-05:00","timeToReady":102153623},{"name":"kubelet","dependencies":["kube-apiserver"],"start":"2024-12-11T10:18:38.349602099-05:00","ready":"2024-12-11T10:18:43.362570424-05:00","timeToReady":5012968316},{"name":"cluster-id-manager","dependencies":["kube-apiserver"],"start":"2024-12-11T10:18:43.363729019-05:00","ready":"2024-12-11T10:18:43.367318249-05:00","timeToReady":3589225}],"microshift":{"start":"2024-12-11T10:18:24.449808837-05:00","servicesStart":"2024-12-11T10:18:24.449808837-05:00","ready":"2024-12-11T10:18:48.365537844-05:00","timeToReady":23915728993}}
Dec 11 10:18:48 el94-base-el94-src-upgrade-ok-host1 microshift[1281]: ??? I1211 10:18:48.366354    1281 run.go:271] sent sd_notify readiness message
Dec 11 10:18:48 el94-base-el94-src-upgrade-ok-host1 microshift[1281]: storage-version-migration-migrator I1211 10:18:48.366469    1281 startuprecorder.go:54] "SERVICE READY" service="storage-version-migration-migrator" since-start="5.00374511s"
```